### PR TITLE
Aarch64: Prevent x86 headers being included on aarch64 builds

### PR DESCRIPTION
**Pull request by jamesmintram**
This PR adds a number of number of commits which prevent includes in the arch/x86 folder from being included in non x86 platforms.

Each header is done as a separate commit to make it clear which changes were required to to make that specific header file work with the new check added.

The fixes involved removing include directives and replacing them with forward declarations.

### CODE REVIEW

File: Kernel/Arch/x86/ASM_wrapper.h [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 #pragma once
 
+#include <AK/Platform.h>
+#if !ARCH(I386) && !ARCH(X86_64)
+#   error "Trying to include x86 only header on non x86 platform"
```
Comment (bgianfo) [inline Kernel/Arch/x86/ASM_wrapper.h:11]: I think we could re-write this as a static_assert, and then put it behind a macro defined in `Platform.h` that all of these files could share? 

Seems like that would be a nice improvement over copying the same logic to all the headers?

File: Kernel/Assertions.h [hunk 2]
```diff
@@ -23,8 +25,13 @@ extern "C" {
...
 }
 
-#define VERIFY_INTERRUPTS_DISABLED() VERIFY(!(cpu_flags() & 0x200))
-#define VERIFY_INTERRUPTS_ENABLED() VERIFY(cpu_flags() & 0x200)
+#if ARCH(I386) || ARCH(X86_64)
+#    define VERIFY_INTERRUPTS_DISABLED() VERIFY(!(cpu_flags() & 0x200))
+#    define VERIFY_INTERRUPTS_ENABLED() VERIFY(cpu_flags() & 0x200)
+#else
+#    define VERIFY_INTERRUPTS_DISABLED() 
+#    define VERIFY_INTERRUPTS_ENABLED()
```
Comment (kleinesfilmroellchen) [inline Kernel/Assertions.h:33]: Can we FIXME this so we know we need to add it later?

File: Kernel/Arch/x86/ASM_wrapper.h (hunk 1 above)
Comment (kleinesfilmroellchen) [reply Kernel/Arch/x86/ASM_wrapper.h:11]: That's a good idea

Comment (jamesmintram) [reply Kernel/Arch/x86/ASM_wrapper.h]: Do you mean something like:

```
#include <AK/platform.h>
ENSURE_ARCH(x86);
```

Comment (bgianfo) [reply Kernel/Arch/x86/ASM_wrapper.h]: Yup!

Comment (jamesmintram) [reply Kernel/Arch/x86/ASM_wrapper.h]: OK, done

File: Kernel/Assertions.h (hunk 2 above)
Comment (jamesmintram) [reply Kernel/Assertions.h]: Added TODO()

File: Kernel/Arch/x86/TrapFrame.h [hunk 3]
```diff
@@ -6,6 +6,9 @@
 
 #pragma once
 
+#include <AK/Platform.h>
+VALIDATE_IS_X86()
```
Comment (bgianfo) [inline Kernel/Arch/x86/TrapFrame.h:10]: nitpick: Can we move these after all of the includes? I think that would fit more with the serenity style.

File: AK/Platform.h [hunk 4]
```diff
@@ -29,6 +29,12 @@
 
 #define ARCH(arch) (defined(AK_ARCH_##arch) && AK_ARCH_##arch)
 
+#if !ARCH(I386) && !ARCH(X86_64)
```
Comment (bgianfo) [inline AK/Platform.h:32]: nit pick: As the reader, it always seems more pleasant to read code "straight threw" than to have to mentally parse negations. Consider inverting the ifdef: 

```cpp
#if ARCH(I386) || ARCH(X86_64)
#    define VALIDATE_IS_X86()
#else
#    define VALIDATE_IS_X86() static_assert(false, "Trying to include x86 only header on non x86 platform");
#endif
```

**Review by bgianfo — APPROVED** [review]
Looks good to me, just some minor nitpicks.

**Review by kleinesfilmroellchen — APPROVED** [review]
I feel like the separate commits for each file is kinda commit farming, though I can't blame you :^). Apart from Brian's picknits, looks good!

**Comment by jamesmintram** [comment]
> I feel like the separate commits for each file is kinda commit farming, though I can't blame you :^). Apart from Brian's picknits, looks good!

Happy to squash them if that is an issue?

From a workflow perspective it's much easier to deal with each in isolation. (Also makes it much easier when editing history)

**Pull request merged**